### PR TITLE
Add support for Numpy 2.0

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

**Comment by arcondello** [comment]
Is missing a release note. Will add.

File: dimod/binary/binary_quadratic_model.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def __len__(self):
         return self.num_variables
 
     def __repr__(self):
-        return "{!s}({!s}, {!s}, {!r}, {!r})".format(type(self).__name__,
+        return "{!s}({!s}, {!s}, {}, {!r})".format(type(self).__name__,
```
Comment (randomir) [inline dimod/binary/binary_quadratic_model.py:295]: fix indent of next line

File: dimod/sampleset.py [hunk 2]
```diff
@@ -1534,7 +1539,7 @@ def aggregate(self):
         # fix the number of occurrences
         record.num_occurrences = 0
         for old_idx, new_idx in enumerate(inverse):
-            record[new_idx].num_occurrences += self.record[old_idx].num_occurrences
+            record["num_occurrences"][new_idx] += self.record[old_idx].num_occurrences
```
Comment (randomir) [inline dimod/sampleset.py:1542]: I'm curious, what's the reason for this change?

**Review by randomir — APPROVED** [review]

File: dimod/sampleset.py (hunk 2 above)
Comment (arcondello) [reply dimod/sampleset.py:1542]: The old one doesn't work anymore. I think it may be a bug in NumPy? I couldn't find a release note or bug report though.
